File: application/backend-credit-card-enrollment/backend-java/src/main/java/cloud/ambar/creditcard/enrollment/projection/isproductactive/ProductActiveStatusRepository.java
package cloud.ambar.creditcard.enrollment.projection.isproductactive;

import cloud.ambar.common.projection.MongoTransactionalProjectionOperator;
import lombok.RequiredArgsConstructor;
import org.springframework.data.mongodb.core.query.Criteria;
import org.springframework.data.mongodb.core.query.Query;
import org.springframework.stereotype.Service;
import org.springframework.web.context.annotation.RequestScope;

import java.util.Optional;

@RequestScope
@Service
@RequiredArgsConstructor
public class ProductActiveStatusRepository {
    private final MongoTransactionalProjectionOperator mongoTransactionalProjectionOperator;

    public boolean isThereAnActiveProductWithId(final String productId) {
        Optional<ProductActiveStatus> productActiveStatus = Optional.ofNullable(mongoTransactionalProjectionOperator.operate().findOne(
                Query.query(
                        Criteria.where("id").is(productId)
                                .and("active").is(true)
                ),
                ProductActiveStatus.class,
                "CreditCard_Enrollment_ProductActiveStatus"
        ));

        return productActiveStatus.isPresent() && productActiveStatus.get().getActive();
    }

    public Optional<ProductActiveStatus> findOneById(final String productId) {
        return Optional.ofNullable(mongoTransactionalProjectionOperator.operate().findOne(
                Query.query(Criteria.where("id").is(productId)),
                ProductActiveStatus.class,
                "CreditCard_Enrollment_ProductActiveStatus"
        ));
    }

    public void save(final ProductActiveStatus productActiveStatus) {
        mongoTransactionalProjectionOperator.operate().save(productActiveStatus, "CreditCard_Enrollment_ProductActiveStatus");
    }
}

File: application/backend-credit-card-enrollment/backend-c#/CreditCardEnrollment/CreditCard/Enrollment/Projection/IsProductActive/ProductActiveStatusRepository.cs
using CreditCardEnrollment.Common.Projection;
using MongoDB.Driver;

namespace CreditCardEnrollment.CreditCard.Enrollment.Projection.IsProductActive;

public class ProductActiveStatusRepository {
    private readonly MongoTransactionalProjectionOperator _mongoOperator;

    public ProductActiveStatusRepository(MongoTransactionalProjectionOperator mongoOperator) {
        _mongoOperator = mongoOperator;
    }

    public bool IsThereAnActiveProductWithId(string productId) {
        return Collection().Find(p => p.Id == productId && p.Active).Any();
    }

    public ProductActiveStatus? FindOneById(string productId) {
        return Collection().Find(p => p.Id == productId).FirstOrDefault();
    }

    public void Save(ProductActiveStatus productActiveStatus) {
        Collection().ReplaceOne(
            p => p.Id == productActiveStatus.Id, 
            productActiveStatus, 
            new ReplaceOptions { IsUpsert = true }
        );
    }

    private IMongoCollection<ProductActiveStatus> Collection() {
        return _mongoOperator.Operate().GetCollection<ProductActiveStatus>("CreditCard_Enrollment_ProductActiveStatus");
    }
}